### PR TITLE
Onboarding Highlights Ship Review 

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1208291253556033/f

**Description**:

This PR includes changes and fixes that have been addressed during the Ship Review.

Commits are mostly self contained and address each Ship Review comment, so you can review each commit separately if you prefer.
The PR also includes the copy translations.

**Steps to test this PR**:
Test Scenario `1`, `2`, and `3` of [Ship Review Testing section](https://app.asana.com/0/1142021229838617/1208350450459625/f)

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/Debouncer.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        timer?.invalidate()
+
+        guard dueTime > 0 else { return block() }
+
+        let timer = Timer(timeInterval: dueTime, repeats: false, block: { timer in
+            guard timer.isValid else { return }
+            block()
+        })
+
+        RunLoop.main.add(timer, forMode: .common)
```
Comment (brindy) [inline Core/Debouncer.swift]: Nothing about this class suggests this is for the main loop.  Can that be more explicit?  Given you've put it in a common location I'd be tempted to also pass in the RunLoop to use, maybe default it to `.current` which is what I would expect as a dev tbh.

File: Core/Debouncer.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A class that provides a debouncing mechanism.
+public final class Debouncer {
```
Comment (brindy) [inline Core/Debouncer.swift:23]: Seems old school to implement this with timers when Combine has one built in but:

1/ Then you'd need to pass in the debounce time in the constructor
2/ It's probably just as many lines of code as this

No change required.

File: DuckDuckGo/OnboardingExperiment/ProgressBarView.swift [hunk 3]
```diff
@@ -144,7 +144,7 @@ struct ProgressBarGradient: View {
                     let nextStep = stepInfo.currentStep < stepInfo.totalSteps ? stepInfo.currentStep + 1 : 1
                     stepInfo = OnboardingProgressIndicator.StepInfo(currentStep: nextStep, totalSteps: stepInfo.totalSteps)
                 }, label: {
-                    Text("Update Progress")
+                    Text(verbatim: "Update Progress")
```
Comment (brindy) [inline DuckDuckGo/OnboardingExperiment/ProgressBarView.swift:147]: Why did this change?  `verbatim` means everyone will see "Update Progress" but shouldn't it be localised?

**Review by brindy — CHANGES_REQUESTED** [review]
Added a note to the asana task with the scenarios in them.  Couple of comments here too.

File: DuckDuckGo/OnboardingExperiment/ProgressBarView.swift (hunk 3 above)
Comment (alessandroboron) [reply DuckDuckGo/OnboardingExperiment/ProgressBarView.swift:147]: Ah this is just a SwiftUI preview I built to test the behaviour of the progress bar. When I tap that button I simulate the progress bar increasing its progress. At translation time I noticed it was translating that string.

File: Core/Debouncer.swift (hunk 1 above)
Comment (alessandroboron) [reply Core/Debouncer.swift]: fair point, I will update this and give the dev ability to choose which RunLoop to use.

Comment (alessandroboron) [reply Core/Debouncer.swift]: updated, I’ve also added the RunLoop.Mode as parameter

File: Core/Debouncer.swift (hunk 2 above)
Comment (alessandroboron) [reply Core/Debouncer.swift:23]: I’m fine to change to use Combine under the hood if we want, but yeah probably the lines of code will be very similar.

**Comment by alessandroboron** [comment]
> Added a note to the asana task with the scenarios in them. Couple of comments here too.

I addressed the PR feedback and replied to your Asana comments. I also fixed other bugs I found during testing. All explained in [Asana task](https://app.asana.com/0/1142021229838617/1208350450459625/f)

**Review by brindy — APPROVED** [review]
Seems to work fine now, except for the bay thing.  Am approving but maybe we'll change eBay?

**Review by brindy — APPROVED** [review]
~~LGTM - tested each variant starts the expected flow~~

**Review by brindy — APPROVED** [review]
~~LGTM - tested each variant starts the expected flow~~

**Pull request merged**